libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/location/NavigationLocationEngine.java

```diff
@@ -0,0 +1,104 @@
+package com.mapbox.services.android.navigation.ui.v5.location;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.location.Location;
+
+import com.mapbox.android.core.location.LocationEngine;
+import com.mapbox.android.core.location.LocationEngineListener;
+import com.mapbox.android.core.location.LocationEnginePriority;
+import com.mapbox.android.core.location.LocationEngineProvider;
+import com.mapbox.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.services.android.navigation.v5.location.MockLocationEngine;
+
+public class NavigationLocationEngine {
```

Comment:
Isn't this name a little consfusing? Maybe it could be `NavigationLocationEngineManager` or something